pkg/kubelet/util.go

```diff
@@ -50,7 +49,7 @@ func InitHealthChecking(k *Kubelet) {
 	// TODO: These should probably become more plugin-ish: register a factory func
 	// in each checker's init(), iterate those here.
 	health.AddHealthChecker(health.NewExecHealthChecker(k))
-	health.AddHealthChecker(health.NewHTTPHealthChecker(&http.Client{}))
+	health.AddHealthChecker(health.NewHTTPHealthChecker(health.Timeout))
```

Comment:
If we are supporting timeouts, why not on exec, too?  Set up an alarm signal before executing...